proposals/0079-allow_commission_decrease_at_any_time.md

```diff
@@ -0,0 +1,88 @@
+---
+simd: '0079'
+title: Allow Commission Decrease at Any Time
+authors:
+  - Bryan Ischo (bryan@ischo.com)
+category: Standard
+type: Core
+status: Draft
+created: 2023-10-26
+feature: https://github.com/solana-labs/solana/issues/29361
+---
+
+## Summary
+
+The commission_updates_only_allowed_in_first_half_of_epoch feature disallows
+commission decrease in the second half of epoch. Given that the purpose of this
+feature was to prevent 'rug pulls' which are accomplished by increasing
+commission at the end of epoch and then decreasing commission at the beginning
+of next epoch, disallowing decreases during the second half of epoch is
+unnecessary.
+
+A feature gate must be added to support this SIMD as all validators' vote
+programs must treat commission change instructions the same or else consensus
+will diverge.
+
+## Motivation
+
+Some validator operators may need to decrease commission in order to satisfy
+their own operational criteria.
+
+As an example, a validaor operator may have a policy whereby any error that
+results in reduced stake account rewards for the epoch, will result in the
+operator choosing to reduce commission to 0% for that epoch to ensure that
+stake accounts are not disadvantaged by that error.  Not being allowed to do
+this in the second half of an epoch is a problem because it would prevent that
+commission change until the next epoch, which will not allow this policy to
+take effect for stake accounts which were de-activating during that epoch.
+
+Allowing commission changes during the second half of epochs also enables
+"reverse rug pulls" where a validator has commission set to 100% during the
+majority of an epoch but reduces it to 0% across the epoch boundary.  It's not
+clear why some validator operators choose to do this, but it does not seem
+harmful, and there does appear to be motivation for some validators to do this
+as it's been done numerous times in the past.
+
+## Alternatives Considered
+
+No alternatives were considered.
+
+## New Terminology
+
+Reverse Rug Pull: Setting commission to 100% for most of an epoch, and to 0%
+only across epoch boundaries.  This has the effect of discouraging third party
+stake to a validator while still charging 0% commission.  This is not
+considered to be a technique harmful to stakers.
+
+## Detailed Design
+
+A feature will be added which, when enabled, must cause all node
+implementations' vote program's set-commission instruction handling to first
+check whether the proposed commission change is a decrease or no change, and if
+so, do not invoke the "only allow commission change in first half of epoch"
+rule.
+
+## Impact
+
+Validators will now be able to decrease commission at any time in the epoch,
+but only increase commission in the first half of epochs (because of the
+commission_updates_only_allowed_in_first_half_of_epoch feature already
+implemented).
```

Comment:
I think this is splitting hairs on the purpose of SIMD sections.  I'm going to leave this as-is.

Reply:
```
## Impact

How will the implemented proposal impacts dapp developers, validators, and core contributors?
```

The text included states how the change will impact validators.  There is no impact to the other categories outside of the truism that "any change could affect anyone".

Reply:
Do we know how many validators have the policy mentioned in the motivation section, or how common this problem is? Something like that would be nice to document as impact, because I agree with Richard that unless we understand that this change does seem "nice to have"

Reply:
Is one good reason not enough?

Reply:
The original feature which prevented changes to the commission in the second half was an over-reach.  It had a goal ("prevent rug pulls") and it created collateral damage by implementing a more restrictive than necessary policy.

It doesn't seem like the best way to have technical discussions to have to provide justification for whether or not the reasons are "good enough".  What you think is important, someone else might not.  It's necessary to include importance in prioritization of effort, but when discussing technical merits of a proposal, I think justification just should be a minor factor, and having "one good reason" should be good enough.

In this case, I've already given a justification of why this feature is important *to me*.

I can imagine other scenarios where it might matter.  Will these sway anyone?  Or is a back and forth discussion about whether "enough reasons have been imagined yet" going to be a part of the process?

- Someone may accidentally set a high commission in the first half of epoch, and not realize it until the second half.  It would be better if they could correct their mistake than if they were prevented from correcting it.

- Someone may set a high commission in the first half of the epoch but change their mind and want to change it back down in the second half.  It would be better if they could make their change rather than being restricted without reason from doing so.

- Someone may receive a command from their government or other authority notifying them that if they accept commission, they are breaking the law and will go to jail.  If this occurs in the second half of an epoch, and they can't set commission to 0, they will have no way to prevent going to jail since their validator already earned vote credits and cannot be prevented otherwise from earning commission (YES this is super far-out, but since the threshold for someone feeling this feature is justified is arbitrary, I have no idea how far out to go here).

- Someone may run a sophisticated commission selection scheme that sets a 100% commission in the first half of the epoch, and then has their stakers vote on reducing it incrementally until they reach a compromise (there would be some other, not disclosed here, reason for stakers to vote for 'some commission' above 0 instead of just 0).  If commission cannot be set in the second half of epochs, this mechanism may not work properly (YES this is super far out; see the previous paragraph for why)

...